micro-rdk/src/esp32/entry.rs

```diff
@@ -9,48 +9,53 @@ use std::{
 };
 
 use crate::common::{
-    app_client::{AppClientBuilder, AppClientConfig},
+    app_client::{AppClient, AppClientError},
     conn::{
         mdns::NoMdns,
         network::Network,
-        server::{ViamServerBuilder, WebRtcConfiguration},
+        server::{TlsClientConnector, ViamServerBuilder, WebRtcConfiguration},
     },
-    entry::RobotRepresentation,
-    grpc_client::GrpcClient,
+    entry::{validate_robot_credentials, RobotRepresentation},
+    grpc_client::GrpcClientError,
     log::config_log_entry,
-    provisioning::storage::RobotCredentials,
     restart_monitor::RestartMonitor,
     robot::LocalRobot,
 };
 
-#[cfg(feature = "data")]
-use crate::common::{data_manager::DataManager, data_store::StaticMemoryDataStore};
-
 use super::{
     certificate::GeneratedWebRtcCertificateBuilder,
     dtls::Esp32DtlsBuilder,
     exec::Esp32Executor,
-    tcp::Esp32Stream,
     tls::{Esp32TLS, Esp32TLSServerConfig},
 };
 
 use async_io::Timer;
 use esp_idf_svc::sys::{settimeofday, timeval};
 
 #[cfg(feature = "provisioning")]
-use crate::common::{
-    grpc::ServerError,
-    provisioning::server::ProvisioningInfo,
-    provisioning::storage::{RobotCredentialStorage, WifiCredentialStorage},
+use crate::{
+    common::{
+        grpc::ServerError,
+        provisioning::server::ProvisioningInfo,
+        provisioning::storage::{RobotConfigurationStorage, WifiCredentialStorage},
+    },
+    proto::app::v1::ConfigResponse,
 };
 
-pub async fn serve_web_inner(
-    robot_creds: RobotCredentials,
+pub async fn serve_web_inner<S>(
```

Comment:
Note that `serve_web_inner` between `esp32` and `native` is now almost identical mod some types. I've filed https://viam.atlassian.net/browse/RSDK-8124 to track that work.